app/screens/channel_info/title/public_private/public_private.tsx

```diff
@@ -25,11 +34,62 @@ const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => ({
     },
 }));
 
+const style = StyleSheet.create({
+    bottomsheet: {
+        flex: 1,
+    },
+});
+
 const PublicPrivate = ({displayName, purpose}: Props) => {
+    const intl = useIntl();
     const theme = useTheme();
+    const managedConfig = useManagedConfig<ManagedConfig>();
+    const {bottom} = useSafeAreaInsets();
+
     const styles = getStyleSheet(theme);
     const publicPrivateTestId = 'channel_info.title.public_private';
 
+    const onCopy = useCallback(() => {
+        Clipboard.setString(purpose!);
+        dismissBottomSheet().then(() => {
+            showSnackBar({barType: SNACK_BAR_TYPE.TEXT_COPIED});
+        });
+    }, [purpose]);
+
+    const handleLongPress = useCallback(() => {
+        if (managedConfig?.copyAndPasteProtection !== 'true') {
+            const renderContent = () => {
+                return (
+                    <View style={style.bottomsheet}>
+                        <SlideUpPanelItem
+                            leftIcon='content-copy'
+                            onPress={onCopy}
+                            testID={`${publicPrivateTestId}.bottom_sheet.copy_purpose`}
+                            text={intl.formatMessage({id: 'channel_info.copy_purpose_text', defaultMessage: 'Copy Purpose Text'})}
+                        />
+                        <SlideUpPanelItem
+                            destructive={true}
+                            leftIcon='cancel'
+                            onPress={() => {
+                                dismissBottomSheet();
+                            }}
+                            testID={`${publicPrivateTestId}.bottom_sheet.cancel`}
+                            text={intl.formatMessage({id: 'mobile.post.cancel', defaultMessage: 'Cancel'})}
+                        />
+                    </View>
+                );
+            };
+
+            bottomSheet({
+                closeButtonId: 'close-mardown-link',
+                renderContent,
+                snapPoints: [1, bottomSheetSnapPoint(2, ITEM_HEIGHT, bottom)],
+                title: intl.formatMessage({id: 'post.options.title', defaultMessage: 'Options'}),
+                theme,
+            });
+        }
+    }, [onCopy, managedConfig?.copyAndPasteProtection]);
```

Comment:
This seems to depend also on the theme, bottom and intl.

Reply:
Done 👍🏻